app/src/ui/app.tsx

```diff
@@ -2250,17 +2250,22 @@ export class App extends React.Component<IAppProps, IAppState> {
         }
 
         const { state: repoState, repository } = selectedState
-        const { pullRequestState } = repoState
-        if (pullRequestState === null) {
+        const { pullRequestState, branchesState } = repoState
+        if (
+          pullRequestState === null ||
+          branchesState.tip.kind !== TipState.Valid
+        ) {
```

Comment:
I wonder if the `popup` object should have a snapshot of all this data in it, instead of having to grab it at render time 🤔 

That also makes me wonder if the `return null` pattern is correct in this method (I see it's used in other `case`s too). What I mean is, when we create the `popup` object it should already have all the info needed for that popup, and therefore there shouldn't be this kind of "invalid popup" situations. But that's just food for thought 😳

Reply:
Hmm.. I suppose and maybe it is worth throwing an error if we somehow how got to this state.. because otherwise the app just does nothing on the user invoking this popup (of course shouldn't happen..). However, I think it would just be putting these checks somewhere else (dispatcher or app-store) also we will need quite a few things from the repoState to pass down into the popup for diffing... 

I will look at this in my PR for introducing the diff view and see if it feels better in the dispatcher or app-store/ then I should know what all needs pass in anyways. I could see it having less checks in (or already checked) in the app-store. Thanks for the input.